com.palantir.typescript/src/com/palantir/typescript/preferences/ProjectPreferenceStore.java

```diff
@@ -227,4 +456,34 @@ private IEclipsePreferences getProjectPreferences() {
 
         return projectScope.getNode(TypeScriptPlugin.ID);
     }
+
+    private static BiMap<String, String> createPreferenceToTsConfigPathMap() {
+        BiMap<String, String> map = HashBiMap.<String, String> create();
+        map.put(IPreferenceConstants.COMPILER_COMPILE_ON_SAVE, "compileOnSave");
+        map.put(IPreferenceConstants.COMPILER_DECLARATION, "compilerOptions.declaration");
+        map.put(IPreferenceConstants.COMPILER_EXPERIMENTAL_DECORATORS, "compilerOptions.experimentalDecorators");
+        map.put(IPreferenceConstants.COMPILER_EMIT_DECORATOR_METADATA, "compilerOptions.emitDecoratorMetadata");
+        map.put(IPreferenceConstants.COMPILER_INLINE_SOURCE_MAP, "compilerOptions.inlineSourceMap");
+        map.put(IPreferenceConstants.COMPILER_INLINE_SOURCES, "compilerOptions.inlineSource");
+        map.put(IPreferenceConstants.COMPILER_JSX, "compilerOptions.jsx");
+        map.put(IPreferenceConstants.COMPILER_MODULE, "compilerOptions.module");
+        map.put(IPreferenceConstants.COMPILER_MODULE_RESOLUTION, "compilerOptions.moduleResolution");
+        map.put(IPreferenceConstants.COMPILER_NO_EMIT_ON_ERROR, "compilerOptions.noEmitOnError");
+        map.put(IPreferenceConstants.COMPILER_NO_FALLTHROUGH_CASES_IN_SWITCH, "compilerOptions.noFallthroughCasesInSwitch");
+        map.put(IPreferenceConstants.COMPILER_NO_IMPLICIT_ANY, "compilerOptions.noImplicitAny");
+        map.put(IPreferenceConstants.COMPILER_NO_IMPLICIT_RETURNS, "compilerOptions.noImplicitReturns");
+        map.put(IPreferenceConstants.COMPILER_NO_LIB, "compilerOptions.noLib");
+        map.put(IPreferenceConstants.COMPILER_OUT_DIR, "compilerOptions.outDir");
+        map.put(IPreferenceConstants.COMPILER_OUT_FILE, "compilerOptions.outFile");
+        map.put(IPreferenceConstants.COMPILER_REMOVE_COMMENTS, "compilerOptions.removeComments");
+        map.put(IPreferenceConstants.COMPILER_SOURCE_MAP, "compilerOptions.sourceMap");
+        map.put(IPreferenceConstants.COMPILER_SUPPRESS_EXCESS_PROPERTY_ERRORS, "compilerOptions.suppressExcessPropertyErrors");
+        map.put(IPreferenceConstants.COMPILER_SUPPRESS_IMPLICIT_ANY_INDEX_ERRORS, "compilerOptions.suppressImplicitAnyIndexErrors");
+        map.put(IPreferenceConstants.COMPILER_TARGET, "compilerOptions.target");
+
+        // TODO : handle files
```

Comment:
done in the last commits, please wait that I fix your feedbacks before the next code review ;)